File: core/src/main/java/com/a65apps/core/interactors/locations/PersonLocationModel.java
package com.a65apps.core.interactors.locations;

import com.a65apps.core.entities.Location;

import java.util.List;

import io.reactivex.rxjava3.core.Single;

public class PersonLocationModel implements PersonLocationInteractor {

    private final PersonLocationRepository personLocationRepository;

    public PersonLocationModel(PersonLocationRepository personLocationRepository) {
        this.personLocationRepository = personLocationRepository;
    }


    @Override
    public Single<Location> loadLocationByPerson(String personId) {
        return personLocationRepository.getLocationByPerson(personId);
    }

    @Override
    public Single<List<Location>> loadAllPersonLocations() {
        return personLocationRepository.getAllPersonLocation();
    }

    @Override
    public Single<Location> createPersonLocation(Location location) {
        return personLocationRepository.createPersonLocation(location);
    }
}

File: core/src/main/java/com/a65apps/core/interactors/locations/PersonLocationModel.kt
package com.a65apps.core.interactors.locations

import com.a65apps.core.entities.Location
import kotlinx.coroutines.flow.Flow

class PersonLocationModel(private val personLocationRepository: PersonLocationRepository) : PersonLocationInteractor {
    override fun loadLocationByPerson(personId: String): Flow<Location> {
        return personLocationRepository.getLocationByPerson(personId)
    }

    override fun loadAllPersonLocations(): Flow<List<Location>> {
        return personLocationRepository.getAllPersonLocation()
    }

    override fun createPersonLocation(location: Location): Flow<Location> {
        return personLocationRepository.createPersonLocation(location)
    }
}